src/components/menu/menu.spec.tsx

```diff
@@ -297,4 +268,32 @@ describe("Menu", () => {
       }
     );
   });
+
+  describe("when a MenuItem unmounts", () => {
+    const MockMenu = () => {
+      const [show, setShow] = React.useState(true);
+
+      return (
+        <Menu>
+          {show && (
+            <MenuItem onClick={() => setShow(false)}>test element 1</MenuItem>
+          )}
+          <MenuItem href="#">test element 2</MenuItem>
+        </Menu>
+      );
+    };
+    it("unregisters itself to the parent Menu", () => {
+      wrapper = mount(<MockMenu />);
+
+      act(() => {
+        wrapper
+          .find(StyledMenuItemWrapper)
+          .first()
+          .props()
+          .onClick({} as React.MouseEvent<HTMLButtonElement>);
+      });
+
+      wrapper.update();
```

Comment:
Are we missing an assertion here to check if the `MenuItem` is still on the page?

Reply:
Good point, I originally added this just to get the last bit of coverage but I seem to have missed an assertion 🤦